File: src/main/java/com/paragon/client/managers/DiscordPresenceManager.java
package com.paragon.client.managers;

import club.minnced.discord.rpc.DiscordEventHandlers;
import club.minnced.discord.rpc.DiscordRPC;
import club.minnced.discord.rpc.DiscordRichPresence;
import com.paragon.Paragon;
import com.paragon.api.util.Wrapper;

public class DiscordPresenceManager implements Wrapper {

    private final String id = "965612502434082846";
    private final DiscordRichPresence presence = new DiscordRichPresence();
    private final DiscordRPC rpc = DiscordRPC.INSTANCE;

    public void startRPC() {
        DiscordEventHandlers eventHandlers = new DiscordEventHandlers();
        rpc.Discord_Initialize(id, eventHandlers, true, null);

        presence.largeImageKey = "logo";
        presence.startTimestamp = System.currentTimeMillis() / 1000L;
        presence.largeImageKey = "Paragon";
        presence.largeImageText = "Paragon Client";
        presence.details = nullCheck() ? "Idling" : mc.isSingleplayer() ? "Singleplayer" : mc.getCurrentServerData() == null ? "pargon client threw a java.lang.NullPointerException" : mc.getCurrentServerData().serverIP;
        presence.state = nullCheck() ? "touching grass" : "beating newfags with paragon client";

        rpc.Discord_UpdatePresence(presence);
    }

    public void updateRPC() {
        presence.largeImageKey = "logo";
        presence.startTimestamp = System.currentTimeMillis() / 1000L;
        presence.largeImageKey = "Paragon";
        presence.largeImageText = "Paragon Client";
        presence.details = nullCheck() ? "Idling" : mc.isSingleplayer() ? "Singleplayer" : mc.getCurrentServerData() == null ? "pargon client threw a java.lang.NullPointerException" : mc.getCurrentServerData().serverIP;
        presence.state = nullCheck() ? "touching grass" : mc.isSingleplayer() ? "trying out paragon in singleplayer" : "beating newfags with paragon client";

        rpc.Discord_UpdatePresence(presence);
    }

    public void stopRPC() {
        rpc.Discord_Shutdown();
        rpc.Discord_ClearPresence();
    }

}

File: src/main/java/com/paragon/client/managers/DiscordPresenceManager.kt
package com.paragon.client.managers

import club.minnced.discord.rpc.DiscordEventHandlers
import club.minnced.discord.rpc.DiscordRPC
import club.minnced.discord.rpc.DiscordRichPresence
import com.paragon.api.util.Wrapper
import com.paragon.api.util.mc

/**
 * @author Wolfsurge
 */
class DiscordPresenceManager : Wrapper {

    private val id = "1000734552047759421"
    private val presence = DiscordRichPresence()
    private val rpc = DiscordRPC.INSTANCE

    fun startRPC() {
        val eventHandlers = DiscordEventHandlers()
        rpc.Discord_Initialize(id, eventHandlers, true, null)

        presence.largeImageKey = "paragon_large"
        presence.startTimestamp = System.currentTimeMillis() / 1000L
        presence.largeImageText = "Paragon Client @ https://github.com/Wolfsurge/Paragon"
        presence.details = getDetails()
        presence.state = "Username: " + mc.session.username
        rpc.Discord_UpdatePresence(presence)
    }

    fun updateRPC() {
        presence.largeImageKey = "paragon_large"
        presence.state = "Username: " + mc.session.username
        presence.startTimestamp = System.currentTimeMillis() / 1000L
        presence.largeImageText = "Paragon Client @ https://github.com/Wolfsurge/Paragon"
        presence.details = getDetails()
        presence.state = "Username: " + mc.session.username
        rpc.Discord_UpdatePresence(presence)
    }

    fun stopRPC() {
        rpc.Discord_Shutdown()
        rpc.Discord_ClearPresence()
    }

    private fun getDetails() = if (nullCheck()) "Idling" else if (minecraft.isSingleplayer) "Playing on singleplayer" else if (minecraft.currentServerData == null) "paragon client threw a java.lang.NullPointerException" else "Playing on " + minecraft.currentServerData!!.serverIP

}